src/games/rcll/net.clp

```diff
@@ -783,3 +786,35 @@
   (pb-broadcast ?peer-id-public ?vi)
   (pb-destroy ?vi)
 )
+
+(deffunction net-create-StorageInfo (?team-color ?mps-ss)
+	(bind ?si (pb-create "llsf_msgs.StorageInfo"))
+	(pb-set-field ?si "team_color" ?team-color)
+	(pb-set-field ?si "machine" ?mps-ss)
+	(do-for-all-facts ((?ssf machine-ss-shelf-slot))
+	                       (eq ?ssf:name ?mps-ss)
+	                  (bind ?ssf-pb (pb-create "llsf_msgs.ShelfSlotInfo"))
+	                  (pb-set-field ?ssf-pb "shelf" (nth$ 1 ?ssf:position))
+	                  (pb-set-field ?ssf-pb "slot" (nth$ 2 ?ssf:position))
+	                  (pb-set-field ?ssf-pb "is_filled" ?ssf:is-filled)
+	                  (pb-set-field ?ssf-pb "description" ?ssf:description)
+	                  (pb-add-list ?si "shelf_slot_info" ?ssf-pb)
+	)
+	(return ?si)
+)
+
+(defrule net-send-StorageInfo
+	(time $?now)
+	?f <- (signal (type storage-info) (time $?t&:(timeout ?now ?t ?*STORAGE-INFO-PERIOD*)) (seq ?seq))
+	(gamestate (cont-time ?ctime) (phase PRODUCTION))
+	(machine (name ?ss-cyan) (mtype SS) (team ?cyan&CYAN))
+	(machine (name ?ss-magenta) (mtype SS) (team ?magenta&MAGENTA))
+	(network-peer (group ?cyan) (id ?peer-id-cyan))
+	(network-peer (group ?magenta) (id ?peer-id-magenta))
+	=>
+	(modify ?f (time ?now) (seq (+ ?seq 1)))
+	(bind ?si-cyan (net-create-StorageInfo ?cyan ?ss-cyan))
+	(bind ?si-magenta (net-create-StorageInfo ?magenta ?ss-magenta))
+	(pb-broadcast ?peer-id-cyan ?si-cyan)
+	(pb-broadcast ?peer-id-magenta ?si-magenta)
+)
```

Comment:
Why is this in a separate message rather than in MachineInfo, as for the other machine types?

Reply:
It would make sense to add the StorageInfo to MachineInfo i guess. I was mainly concerned with sending an update on all 48 storage slots every 0.25 seconds. In our simulation this is no issue, so maybe we can indeed put the storage info inside the machine info.